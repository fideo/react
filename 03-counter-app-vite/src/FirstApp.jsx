//rafc para crear el reactArrowFunctionComponent
//import { Fragment } from 'react'; //esto se reemplaza con <>
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const getMessage = (usuario) =>{
  return "Hola " + usuario +", bienvenido!!!";
}

const newMessage = {
    message: 'Hola Mundo',
    name: 'Federico'
}

export const FirstApp = ( { title, subtitle, name } ) => {

  //console.log(props);
  if (!title){
    throw new Error('El title no existe');
  }

  return (
    // <Fragment>
    //esto <> reemplaza al <Fragment>
    <> 
      {getMessage(name)}
      <h1>{ title }</h1>
        {/* <code>{ JSON.stringify( newMessage ) }</code> */}
      <p>{ subtitle }</p>
    </>
    // </Fragment>
  )
}

FirstApp.propTypes = {
  subtitle: PropTypes.string,
  title: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
  name: 'Federico Mazzei',
  subtitle: 'No hay subtitulo',
  title: 'No hay titulo',
}