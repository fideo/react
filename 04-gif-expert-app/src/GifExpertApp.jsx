//rafc Para crear un funcional component
import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = () => {

  const [categories, setCategories] = useState([ 'One Punch' ]);

  const onAddCategory = ( newCategory ) => {
    //TODO resolver el ingresos CaseSensitive .toLowerCase() comparando 
    if (categories.includes(newCategory)) return;

    setCategories( [ newCategory, ...categories ] );

  }

  return (
    <>
    {/* titulo */}
      <h1>GifExpertApp</h1>

    {/* Input */}
    <AddCategory 
      // setCategories={ setCategories }
      onNewCategory={ (value) => onAddCategory(value)  }
     />

    {/* Listado de Gif */}
    { 
        categories.map( ( category ) => (
          <GifGrid 
              key={ category } 
              category={ category } />
        ))
    }
      {/* <li>ABC</li>
      <li>123</li> */}
      {/* Gif Item */}

    </>
  )
}
