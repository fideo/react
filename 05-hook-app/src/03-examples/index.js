export * from './Character'
export * from './LoadingCharacter'
export * from './MultipleCustomHooks'