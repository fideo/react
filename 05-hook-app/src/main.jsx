import React from "react";
import ReactDOM from "react-dom/client";
import { FormWithCustomHook } from "./02-useEffect/FormWithCustomHook";
import { SimpleForm } from "./02-useEffect/SimpleForm";
import { MultipleCustomHooks } from "./03-examples/MultipleCustomHooks";
import { FocusScreen } from "./04-useRef/FocusScreen";
import { Layout } from "./05-useLayoutEffect/Layout";
import { CallbackHooks } from "./06-memos/CallbackHooks";
import { MemoHook } from "./06-memos/MemoHook";
import { Memorize } from "./06-memos/Memorize";
import { Padre } from "./07-tarea-memo/Padre";
//import { CounterApp } from "./01-useState/CounterApp";
//import { CounterWithCustomHook } from "./01-useState/CounterWithCustomHook";
import { HooksApp } from "./HooksApp";
//import './08-useReducer/intro-reducer'
import { TodoApp } from "./08-useReducer/TodoApp";

import "./index.css";
import { MainApp } from "./09-useContext/MainApp";
import { BrowserRouter } from "react-router-dom";


ReactDOM.createRoot(document.getElementById("root")).render(
    //<SimpleForm />
    // <FormWithCustomHook />
    //<MultipleCustomHooks />
    //<FocusScreen />
    // <Layout />
    // <Memorize />
    // <MemoHook />
    // <CallbackHooks />
    //<Padre />
    //<TodoApp />
  <BrowserRouter>
    {/* <React.StrictMode> */}
      <MainApp />
    {/* </React.StrictMode> */}
  </BrowserRouter>

);
