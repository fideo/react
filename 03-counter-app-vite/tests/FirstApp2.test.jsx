import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en <FirstApp2 />', () => { 

  const title = 'Hola, soy Goku';
  const subtitle = 'Hola soy un subtitulo';

  test('debe de hacer match con el snapshot', () => { 
    
      const {container} = render( <FirstApp title={title} />);
      expect( container ).toMatchSnapshot();

   });

  test('Debe de mostrar el mensaje "Hola, Soy Goku"', () => { 
    
      render(<FirstApp title={title} />);
      //screen.debug(); // para ver el objeto;
      expect(screen.getByText(title)).toBeTruthy();
    
   });

   test('debe de mostrar el titulo en h1', () => { 
    
      render(<FirstApp title={title} />);

      expect( screen.getByRole('heading', { level: 1 } ).innerHTML ).toContain(title);

    });


    test('debe de mostrar el subtitulo enviado por props', () => { 
    
      render(<FirstApp title={title} subtitle={subtitle} />);

      expect( screen.getAllByText(subtitle).length ).toBe(2);

    });


});