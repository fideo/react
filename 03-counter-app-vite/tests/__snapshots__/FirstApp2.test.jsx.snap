// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp2 /> debe de hacer match con el snapshot 1`] = `
<div>
  Hola Federico Mazzei, bienvenido!!!
  <h1
    data-testid="test-title"
  >
     
    Hola, soy Goku
     
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
</div>
`;
